py_to_proto/dataclass_to_proto.py

```diff
@@ -251,12 +251,29 @@ def get_oneof_fields(
         if get_origin(field_type) is Union:
             for arg in get_args(field_type):
                 oneof_field_name = self._get_unique_annotation(arg, OneofField)
-                if oneof_field_name is None:
-                    res_type = self._resolve_wrapped_type(arg)
+                res_type = self._resolve_wrapped_type(arg)
+                if get_origin(arg) is list:
+                    assert len(get_args(arg)) > 0, "List does not have any arguments"
+                    field_type = get_args(arg)[0]
+                    oneof_field_name = (
+                        f"{field_def.name}_{str(field_type.__name__)}_sequence".lower()
+                    )
+                    arg = dataclasses.make_dataclass(
+                        f"{str(field_type.__name__).capitalize()}Sequence",
+                        [("values", List[field_type])],
+                    )
+                elif oneof_field_name is None:
                     oneof_field_name = (
                         f"{field_def.name}_{str(res_type.__name__)}".lower()
                     )
                     log.debug3("Using default oneof field name: %s", oneof_field_name)
+                elif get_origin(res_type) is list:
+                    # get the arg from the list
+                    field_type = get_args(get_args(arg)[0])[0]
+                    arg = dataclasses.make_dataclass(
+                        f"{str(field_type.__name__).capitalize()}Sequence",
+                        [("values", List[field_type])],
+                    )
```

Comment:
I _think_ you can simplify it down to this, right?

```suggestion
                res_type = self._resolve_wrapped_type(arg)
                if get_origin(res_type) is list:
                    assert get_args(res_type), "List does not have any arguments"
                    field_type = get_args(res_type)[0]
                    oneof_field_name = oneof_field_name or (
                        f"{field_def.name}_{str(field_type.__name__)}_sequence".lower()
                    )
                    arg = dataclasses.make_dataclass(
                        f"{str(field_type.__name__).capitalize()}Sequence",
                        [("values", List[field_type])],
                    )
                if oneof_field_name is None:
                    oneof_field_name = (
                        f"{field_def.name}_{str(res_type.__name__)}".lower()
                    )
                    log.debug3("Using default oneof field name: %s", oneof_field_name)
```

Reply:
Whoops, I pushed a refactor at the same time!

Reply:
There's a bit of a nuance for handling annotated oneofs, which require a one-level deeper `get_args` within a `get_args` call to actually get the `field_type`.

Reply:
`field_type = get_args(res_type)[0]` this will give a `typing.List[str]` if the field is using `Annotated`, hence we will have to go one level deeper in that case.